torch/nn/modules/module.py

```diff
@@ -1065,9 +1069,37 @@ def eval(self):
         particular modules for details of their behaviors in training/evaluation
         mode, if they are affected, e.g. :class:`Dropout`, :class:`BatchNorm`,
         etc.
+
+        This is equivalent with :meth:`self.train(False) <torch.nn.Module.train>`.
+
+        Returns:
+            Module: self
         """
         return self.train(False)
 
+    def requires_grad_(self, requires_grad=True):
+        r"""Change if autograd should record operations on parameters in this
+        module.
+
+        This method sets the parameters' :attr:`requires_grad` attributes
+        in-place.
+
+        This method is helpful for freezing part of the module for finetuning
+        and training parts of a model individually (e.g., GAN training).
+
+        Args:
+            requires_grad (bool): whether autograd should record operations on
+                                  parameters in this module. Default: ``True``.
+
+        Returns:
+            Module: self
+        """
+        for p in self.parameters():
+            p.requires_grad_(requires_grad)
+        for module in self.children():
+            module.requires_grad_(requires_grad)
```

Comment:
`self.parameters()` iterates over children, so you don't have to do that (right now it all runs in O(n^2) time).

Reply:
thanks for pointing this out! fixed!